src/arrayMethodSort.js

```diff
@@ -4,8 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const conditionFunction =
+      compareFunction ||
+      function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA === strB) {
+          return 0;
+        }
+
+        return strA > strB ? 1 : -1;
+      };
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
```

Comment:
You are currently using the Bubble sort algorithm, which has a time complexity of O(n^2). For large arrays, this can be quite slow. Consider implementing a more efficient sorting algorithm, such as QuickSort or MergeSort. However, note that this is just a suggestion for improvement and not a requirement for this task.